azurelinuxagent/distro/default/osutil.py

```diff
@@ -261,6 +261,9 @@ def set_selinux_context(self, path, con):
         Returns exit result.
         """
         if self.is_selinux_system():
+            if not os.path.exists(path):
```

Comment:
These methods are called from many locations (especially chmod), and adding checks everywhere is both messy and error-prone since any future calls may not enforce those checks. As there is no side-effect from the additional code, but rather the unwanted side-effect of failure is avoided, I feel the change is reasonable.